### PR TITLE
Remove a bunch of old UI cruft

### PR DESCRIPTION
**Pull request by jerith**
There are a bunch of things that were used by the old UI that are no longer called anywhere and can be removed. This work follows on from #671 and friends.

### CODE REVIEW

File: go/vumitools/api.py [hunk 1]
```diff
@@ -52,19 +52,6 @@ class TagpoolSet(object):
...
-
-        :param function filter_func:
-            A function f(pool, metadata) that should return True if the
-            pool should be kept and False if it should be discarded.
-        """
-        new_pools = dict((pool, metadata)
-                         for pool, metadata in self._pools.iteritems()
-                         if filter_func(pool, metadata))
-        return self.__class__(new_pools)
-
```
Comment (hodgestar) [inline go/vumitools/api.py:67]: Not really sure why this is being deleted? Seems useful?

**Comment by hodgestar** [comment]
Other than the question about `select`, :+1:.

File: go/vumitools/api.py (hunk 1 above)
Comment (jerith) [inline go/vumitools/api.py:67]: The only thing that called it was the old conversation form that was removed. We can always bring it back (or write a similar thing) if we need it again.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**